### PR TITLE
Contract delivery manager view

### PR DESCRIPTION
**Pull request by Nimphal**
### What changed

Adds check if the user is a contract delivery manager and if so, loads all available locations.

### Why did it change

Contract delivery managers need an overview of the everything that's happening.
### Issue tracking

- [P4-2639](https://dsdmoj.atlassian.net/browse/P4-2639)

## Screenshots

| Before | After |
| ------ | ----- |
| <kbd></kbd> | <kbd></kbd> |

#### Automated testing

- [x] Added unit tests to cover changes
- [ ] Added end-to-end tests to cover any journey changes

#### Manual testing

Has been tested in the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [x] No environment variables were added or changed

### Other considerations

Commit messages with a `fix` or `feat` type are automatically used to generate the [changelog](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/CHANGELOG.md) and
[GitHub release notes](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/releases) during the release task. Please make sure they will read well on their own in a
summary of changes and that the commit body gives a more detailed description of those changes if necessary.

- [ ] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics
- [ ] Update [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md) with any new instructions or tasks

### CODE REVIEW

File: common/lib/permissions.js [hunk 1]
```diff
@@ -25,6 +25,18 @@ const policePermissions = [
   'person_escort_record:confirm',
 ]
 
+const contractDeliveryManagerPermissions = [
+  'allocations:view',
+  'contract_delivery_manager',
```
Comment (merlinc) [inline common/lib/permissions.js:30]: The pattern we've got for permissions is `area:permission` or `area:sub:permission` - would something like `locations:contract_delivery_manager` make any sense here? Just seeing `contract_delivery_manager` on its own looks weird compared to the other permissions

File: app/locations/controllers.js [hunk 2]
```diff
@@ -24,6 +24,21 @@ async function locations(req, res, next) {
     req.session.user.locations = userLocations
   }
 
+  if (userPermissions.includes('contract_delivery_manager')) {
```
Comment (merlinc) [inline app/locations/controllers.js:27]: If the user has a supplier ID and the `contract_delivery_manager` does that make the call above uneeded?

Comment (Nimphal) [reply app/locations/controllers.js:27]: They won't have supplier Ids, those come from the group (so Serco PECS or Geoamey PECS), and CDMs won't have that group. I don't know if it's worth complicating the per supplier call with a check for CDMs, there are very few CDMs and the chance of them being accidentally added to a supplier group is even lower.

File: common/lib/permissions.js [hunk 3]
```diff
@@ -25,6 +25,18 @@ const policePermissions = [
   'person_escort_record:confirm',
 ]
 
+const contractDeliveryManagerPermissions = [
+  'allocations:view',
```
Comment (teneightfive) [inline common/lib/permissions.js:29]: Do we want to give them `dashboard:view` access too?

**Review by teneightfive — APPROVED** [review]

File: common/lib/permissions.js (hunk 3 above)
Comment (Nimphal) [reply common/lib/permissions.js:29]: The ticket says just moves and stuff for now, and dashboard later, and I suspect we will be tweaking these permissions when people actually have a play anyway. I think I'll merge it like this, have product have a play and then update with whatever is missing.

**Pull request merged**